components/UserObserver.php

```diff
@@ -163,6 +167,11 @@ private function saveUser(User $user, UserAuthentication $user_authentication, a
 			throw new Exception('Unable to save user: '.print_r($user->getErrors(),true));
 		}
 
+		if (!$user_authentication->save(false)) {
+			\Yii::log('Unable to save user: '.print_r($user->getErrors(),true), \CLogger::LEVEL_ERROR);
+			throw new Exception('Unable to save user: '.print_r($user->getErrors(),true));
+		}
```

Comment:
just notice that the User and UserAuthentication records are saved without validation, which probably should be fine, but in `saveContact`, the contact will be saved with validation, which potentially will cause the saved user record does not have an associated contact info (first name, last name, etc.)

Reply:
Validation has been added. The contact_id field is not mandatory.